src/components/KTruncate/KTruncate.vue

```diff
@@ -0,0 +1,281 @@
+<template>
+  <div
+    ref="wrapper"
+    class="k-truncate"
+    :class="[expanded ? 'expanded' : '', `k-truncate-${isTextContent ? 'text' : 'content'}`]"
+  >
+    <div
+      ref="container"
+      class="container"
+    >
+      <slot name="default" />
+      <div
+        v-if="!isTextContent && expanded"
+        data-testid="collapse-trigger-wrapper"
+      >
+        <slot
+          :collapse="handleToggleClick"
+          name="collapse-trigger"
+        >
+          <KButton
+            appearance="primary"
+            class="collapse-trigger"
+            is-rounded
+            @click="handleToggleClick"
+          >
+            <KIcon
+              color="var(--KTruncateCollapseIconColor, var(--blue-500, color(blue-500)))"
+              icon="chevronUp"
+              size="10"
+              title="Show less"
+            />
+          </KButton>
+        </slot>
+      </div>
+    </div>
+    <div
+      v-if="!isTextContent && showToggle"
+      class="d-flex h-100 align-items-end"
+    >
+      <div
+        v-if="!expanded"
+        data-testid="expand-trigger-wrapper"
+      >
+        <slot
+          :excess-elements-count="excessElementsCount"
+          :expand="handleToggleClick"
+          name="expand-trigger"
+        >
+          <KButton
+            appearance="btn-link"
+            class="expand-trigger"
+            @click="handleToggleClick"
+          >
+            {{ excessElementsCount }}
+          </KButton>
+        </slot>
+      </div>
+    </div>
+    <div
+      v-if="isTextContent && (showToggle || expanded)"
+      ref="textToggleControls"
+      class="place-self-end mt-2"
+    >
+      <div
+        v-if="!expanded"
+        data-testid="expand-trigger-wrapper"
+      >
+        <slot
+
+          :expand="handleToggleClick"
+          name="expand-trigger"
+        >
+          <KButton
+            appearance="btn-link"
+            class="expand-trigger"
+            @click="handleToggleClick"
+          >
+            Show more
+          </KButton>
+        </slot>
+      </div>
+      <div v-if="expanded">
+        <slot
+          :collapse="handleToggleClick"
+          name="collapse-trigger"
+        >
+          <KButton
+            appearance="btn-link"
+            class="collapse-trigger"
+            @click="handleToggleClick"
+          >
+            Show less
+          </KButton>
+        </slot>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { onMounted, onBeforeUnmount, ref, nextTick } from 'vue'
+
+const props = defineProps({
+  rows: {
+    type: Number,
+    default: 1,
+    validator: (value: Number) => value > 0,
+  },
+  isTextContent: {
+    type: Boolean,
+    default: false,
+  },
```

Comment:
I'm a little confused on the prop name, but also on why this prop is necessary. 

You can slot a tag (e.g. `<p>`) with text content, so really HTML and text are valid.

Reply:
This prop is necessary because by default the component treats the content as collection of HTML elements and applies more complex logic. When this prop is true, that logic is skipped, instead -webkit-line-clamp property is applied. What do you think would be a good name?

Reply:
Can you pass in multiple `<p>` tags and still allow truncation, or when this prop is true can there only be a single child? Regardless of the answer, I think we need to add more details in the documentation for the prop.

Reply:
Yes, you can pass multiple `<p>` tags, doesn't have to be single child. I added another paragraph explaining this logic, plz take a look and let me know if you think we need anything else